lib/poseidon/sync_producer.rb

```diff
@@ -57,6 +57,7 @@ def send_messages(messages)
         if !messages_to_send.pending_messages? || @max_send_retries == 0
           break
         else
+          Poseidon.logger.debug { "retrying sending #{messages_to_send.messages} messages"  }
```

Comment:
Yeah, I kinda suspect whatever savings there is from this is probably a wash with instantiating blocks, which aren't cheap either. Mostly, I'm just trying to stick with project norms.